image_services/egis/make_egis_services.py

```diff
@@ -0,0 +1,329 @@
+        'Tag': tag,
+        'GroupName': tag,
+        'StartDate': SEASONS[season]['StartDate'],
+        'EndDate': SEASONS[season]['EndDate'],
+        'ProductType': product_type,
+        'Season': SEASONS[season]['Season'],
+        'Polarization': polarization,
+        'Tile': tile,
+        'DownloadURL': download_url,
+        'URLDisplay': name,
+        'MaxPS': '910',
+    }
+
+
+def update_csv(csv_file: str, rasters: List[str]):
+    if os.path.isfile(csv_file):
+        with open(csv_file) as f:
+            records = [record for record in csv.DictReader(f)]
+    else:
+        records = []
+    logging.info(f'Found {len(records)} items in {csv_file}')
+
+    existing_rasters = [record['Raster'] for record in records]
+    new_rasters = set(rasters) - set(existing_rasters)
+
+    logging.info(f'Adding {len(new_rasters)} new items to {csv_file}')
+    for raster in new_rasters:
+        record = get_raster_metadata(raster)
+        records.append(record)
+
+    records = sorted(records, key=lambda x: x['Raster'])
+    with open(csv_file, 'w', newline='') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=records[0].keys(), lineterminator=os.linesep)
+        writer.writeheader()
+        writer.writerows(records)
+
+
+def add_overviews(mosaic_dataset, local_path):
+    # This function adds CRF overviews to the mosaic dataset and calculates custom attribute values
+    print('Calculating field values for overview record')
+    ds = os.path.join(local_path, mosaic_dataset)
+    ds_cursor = arcpy.da.UpdateCursor(ds, ["Tag", "MinPS", "Category", "StartDate", "EndDate", "GroupName",
+                                           "Name", "ProductType", "Season", "Polarization", "Tile", "DownloadURL",
+                                           "URLDisplay"])
+    if ds_cursor is not None:
+        print('Updating Overview Field Values')
+        for row in ds_cursor:
+            try:
+                NameOvField = row[6]
+                ProdTypeOvField = NameOvField.split("_")[1]
+                SeasonOvCode = NameOvField.split("_")[3]
+                if SeasonOvCode == 'DJF':
+                    SeasonOvField = 'December/January/February'
+                elif SeasonOvCode == 'MAM':
+                    SeasonOvField = 'March/April/May'
+                elif SeasonOvCode == 'JJA':
+                    SeasonOvField = 'June/July/August'
+                elif SeasonOvCode == 'SON':
+                    SeasonOvField = 'September/October/November'
+                else:
+                    SeasonOvField = 'unknown'
+                PolOvField = NameOvField.split("_")[2]
+                TileOvField = 'Zoom in further to see specific tile information'
+                DLOvField = 'Zoom in further to access download link'
+                if row[0] == 'Dataset':
+                    row[1] = 900
+                    row[2] = 2
+                    row[3] = '12/01/2019'
+                    row[4] = '11/30/2020'
```

Comment:
@hjkristenson We set the start/end dates for the geotiffs to the start/end dates of the season. Do we want the start/end dates for the overview to also be the start/end dates of the specific season, or of the whole year?